src/hooks/useSignUp.js

```diff
@@ -0,0 +1,22 @@
+import { app } from "../utils/firebaseConfig";
+
+import { getAuth, createUserWithEmailAndPassword } from "firebase/auth";
+
+const auth = getAuth(app);
+
+export const useSignUp = async (email, password) => {
```

Comment:
Here we can do a validation before creating the new user in case the account already exists, this way we can show a message to the user about the email address already being used. An alert like the one for "incorrect password"